Perlang.Interpreter/PerlangInterpreter.cs

```diff
@@ -365,6 +372,12 @@ private void RegisterGlobalClasses()
                     GetVariableOrFunctionBinding
                 );
 
+                // All validation was successful, but unlike for statements, there is no need to mutate the
+                // previousStatements field in this case. Think about it for a moment. We know that the line being
+                // interpreted is an expression, so it _cannot_ have declared any new variable or anything like that
+                // (those are only allowed in statements). Hence, we presume that this expression is, if you will,
+                // "side-effect-free" in that sense.
+
```

Comment:
The hunks for this file is the critical part of this PR. We simply _must_ use the information in the previous statement, together with the expression given in the new statement to be able to properly perform all validations. :point_up: is where the bug was that caused this problem.